libs/designsystem/src/lib/components/progress-circle/progress-circle.component.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  Component,
+  Input,
+  ChangeDetectionStrategy,
+  HostBinding,
+  ElementRef,
+  AfterViewInit,
+  ChangeDetectorRef,
+  OnDestroy,
+} from '@angular/core';
+
+const RADIUS_MAP = {
+  sm: 20,
+  md: 28,
+  lg: 48,
+};
+
+@Component({
+  selector: 'kirby-progress-circle',
+  templateUrl: './progress-circle.component.html',
+  styleUrls: ['./progress-circle.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ProgressCircleComponent implements AfterViewInit, OnDestroy {
+  @Input() value: number = 0;
+  @Input() size: 'sm' | 'md' | 'lg' = 'md';
+  @Input() themeColor: 'success' | 'warning' | 'danger';
+
+  private hasElementBeenVisible: boolean | undefined;
+  private observer: IntersectionObserver;
+
+  constructor(private elementRef: ElementRef, private changeDetectorRef: ChangeDetectorRef) {}
+
+  ngAfterViewInit() {
+    this.observer = new IntersectionObserver(this.onElementVisible, {
+      threshold: 0.5,
+    });
+    this.observer.observe(this.elementRef.nativeElement);
+  }
+
+  ngOnDestroy(): void {
+    this.unobserve();
+  }
+
+  private onElementVisible = (entries: IntersectionObserverEntry[]) => {
+    this.hasElementBeenVisible = entries && entries.some((entry) => entry.isIntersecting);
+    if (this.hasElementBeenVisible) {
+      this.unobserve();
+      this.changeDetectorRef.markForCheck();
+    }
+  };
+
+  private unobserve() {
+    if (this.observer) {
+      this.observer.unobserve(this.elementRef.nativeElement);
+      // Safari does not support "disconnect", see: https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API#Browser_compatibility
+      if (typeof this.observer.disconnect === 'function') {
+        this.observer.disconnect();
+      }
+    }
+  }
+
+  @HostBinding('style.width.px')
+  @HostBinding('style.height.px')
+  get diameter(): number {
+    return this.radius * 2;
```

Comment:
Consider swapping out with a `DIAMETER_MAP` as it's more in line with UX sizes:
```suggestion
    return DIAMETER_MAP[this.size];
```

Reply:
Changed to diameter map